snapcraft/parts/lifecycle.py

```diff
@@ -302,7 +302,7 @@ def _run_in_provider(
     with provider.launched_environment(
         project_name=project.name,
         project_path=Path().absolute(),
-        base=cast(str, project.base),
+        base=project.get_effective_base(),
```

Comment:
nice name, but I would call it get_build_base, if set,it is build-base, if not the base; but in all cases, it is the base where things are built on.